File: sample/src/main/java/com/schibsted/spain/barista/sample/DialogActivity.java
package com.schibsted.spain.barista.sample;

import android.content.DialogInterface;
import android.os.Bundle;
import androidx.appcompat.app.AlertDialog;
import androidx.appcompat.app.AppCompatActivity;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

public class DialogActivity extends AppCompatActivity {

  @Override
  protected void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    setContentView(R.layout.activity_dialog);

    final TextView textView = (TextView) findViewById(R.id.dialog_selected_button_value);

    final AlertDialog.Builder dialog = new AlertDialog.Builder(this)
        .setTitle("Dialog")

        .setPositiveButton("positive", new DialogInterface.OnClickListener() {
          @Override
          public void onClick(DialogInterface dialog, int which) {
            textView.setText("positive");
          }
        })
        .setNegativeButton("negative", new DialogInterface.OnClickListener() {
          @Override
          public void onClick(DialogInterface dialog, int which) {
            textView.setText("negative");
          }
        })
        .setNeutralButton("neutral", new DialogInterface.OnClickListener() {
          @Override
          public void onClick(DialogInterface dialog, int which) {
            textView.setText("neutral");
          }
        });

    Button button = (Button) findViewById(R.id.button);
    button.setOnClickListener(new View.OnClickListener() {
      @Override
      public void onClick(View v) {
        dialog.create().show();
      }
    });
  }
}

File: sample/src/main/java/com/schibsted/spain/barista/sample/DialogActivity.kt
package com.schibsted.spain.barista.sample

import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.TextView
import androidx.appcompat.app.AlertDialog
import androidx.appcompat.app.AppCompatActivity

class DialogActivity : AppCompatActivity() {
  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)
    setContentView(R.layout.activity_dialog)
    val textView = findViewById<View>(R.id.dialog_selected_button_value) as TextView
    val dialog = AlertDialog.Builder(this)
      .setTitle("Dialog")
      .setPositiveButton("positive") { dialog, which -> textView.text = "positive" }
      .setNegativeButton("negative") { dialog, which -> textView.text = "negative" }
      .setNeutralButton("neutral") { dialog, which -> textView.text = "neutral" }
    val button = findViewById<View>(R.id.button) as Button
    button.setOnClickListener { dialog.create().show() }
  }
}